src/pages/map.tsx

```diff
@@ -112,17 +135,49 @@ const MapPage = ({ schedule }: Props) => {
     lokasi: item
   }));
 
+  const setInitialMapBound = () => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    if ('permissions' in navigator) {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      window.navigator.permissions.query({ name: 'geolocation' }).then(data => {
+        const permission = data.state === 'granted';
+        setGetGeoPermission(permission);
+      });
+    }
+  };
+
+  React.useEffect(() => {
+    const listOfCoordinate = coordinates.map(data => ({
+      lat: Number(data.lat),
+      lng: Number(data.lng)
+    }));
+    if (map) {
+      if (geoObj?.lat && geoObj.lng) {
+        listOfCoordinate.push({ lat: geoObj.lat, lng: geoObj.lng });
+
+        map.fitBounds(getMapBounds(listOfCoordinate), { padding: 100 });
+      } else {
+        map.fitBounds(getMapBounds(coordinates as LngLat[]), { padding: 100 });
```

Comment:
not sure why you don't reuse `listOfCoordinate` here, did I miss something?

Reply:
sorry im drunk...done